compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1045,6 +1047,78 @@ pub fn prohibit_assoc_item_binding(
             None
         },
     });
+
+    // Emit a suggestion if possible
+    if let Some((def_id, segment, _)) = segment
+        && segment.args().parenthesized == hir::GenericArgsParentheses::No
+        && let hir::TypeBindingKind::Equality { term: binding_arg_ty } = binding.kind
+    {
+        // Suggests removal of the offending equality constraint
+        let suggest_removal = |e: &mut Diag<'_>| {
+            let mut suggestion_span = binding.span.with_hi(binding.span.hi() + BytePos(1)); // Include the comma or the angle bracket at the end
```

Comment:
Will do that.

Reply:
I've made the span calculation smarter now. It extends now to the `hi` of the preceding arg and `lo` of the following arg as needed instead of extending by a hard coded value of `BytePos(1)`